quinn-proto/src/endpoint.rs

```diff
@@ -200,11 +201,13 @@ impl Endpoint {
             }
         };
 
-        // Handle packet on existing connection, if any
-
         let addresses = FourTuple { remote, local_ip };
+        let dst_cid = event.first_decode.dst_cid();
+
         if let Some(route_to) = self.index.get(&addresses, &event.first_decode) {
-            return match route_to {
+            // Handle packet on existing connection, if any
```

Comment:
In this branch, we know there's an existing connection [attempt], so "if any" no longer makes sense.